Coverage job is failing for knative/build-pipeline

There's an error in the job that's causing coverage to fail. Latest daily coverage job failed with (note that the job succeeds even though the coverage failed):

https://prow.knative.dev/log?job=ci-knative-build-pipeline-go-coverage&id=1063597766501994499

(parsed the stacktrace manually)

```
Error running 'go test -coverprofile /logs/artifacts/coverage_profile.txt'

cmd/controller/main.go:38:2: cannot find package "github.com/knative/build-pipeline/pkg/client/clientset/versioned" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/client/clientset/versioned (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/client/clientset/versioned (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/client/clientset/versioned (from $GOPATH)
cmd/controller/main.go:39:2: cannot find package "github.com/knative/build-pipeline/pkg/client/informers/externalversions" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/client/informers/externalversions (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/client/informers/externalversions (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/client/informers/externalversions (from $GOPATH)
cmd/controller/main.go:29:2: cannot find package "github.com/knative/build-pipeline/pkg/logging" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/logging (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/logging (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/logging (from $GOPATH)
cmd/controller/main.go:31:2: cannot find package "github.com/knative/build-pipeline/pkg/reconciler" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/reconciler (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/reconciler (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/reconciler (from $GOPATH)
cmd/controller/main.go:32:2: cannot find package "github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/pipelinerun" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/pipelinerun (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/pipelinerun (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/pipelinerun (from $GOPATH)
cmd/controller/main.go:33:2: cannot find package "github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun (from $GOPATH)
cmd/controller/main.go:34:2: cannot find package "github.com/knative/build-pipeline/pkg/system" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/system (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/system (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/system (from $GOPATH)
cmd/kubeconfigwriter/main.go:30:2: cannot find package "github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1 (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1 (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1 (from $GOPATH)
pkg/apis/pipeline/v1alpha1/register.go:20:2: cannot find package "github.com/knative/build-pipeline/pkg/apis/pipeline" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/apis/pipeline (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/apis/pipeline (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/apis/pipeline (from $GOPATH)
pkg/client/clientset/versioned/clientset.go:19:2: cannot find package "github.com/knative/build-pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1 (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1 (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1 (from $GOPATH)
pkg/client/clientset/versioned/fake/clientset_generated.go:21:2: cannot find package "github.com/knative/build-pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1/fake" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1/fake (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1/fake (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1/fake (from $GOPATH)
pkg/client/clientset/versioned/typed/pipeline/v1alpha1/pipeline.go:20:2: cannot find package "github.com/knative/build-pipeline/pkg/client/clientset/versioned/scheme" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/client/clientset/versioned/scheme (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/client/clientset/versioned/scheme (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/client/clientset/versioned/scheme (from $GOPATH)
pkg/client/informers/externalversions/factory.go:24:2: cannot find package "github.com/knative/build-pipeline/pkg/client/informers/externalversions/internalinterfaces" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/client/informers/externalversions/internalinterfaces (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/client/informers/externalversions/internalinterfaces (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/client/informers/externalversions/internalinterfaces (from $GOPATH)
pkg/client/informers/externalversions/factory.go:25:2: cannot find package "github.com/knative/build-pipeline/pkg/client/informers/externalversions/pipeline" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/client/informers/externalversions/pipeline (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/client/informers/externalversions/pipeline (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/client/informers/externalversions/pipeline (from $GOPATH)
pkg/client/informers/externalversions/pipeline/interface.go:20:2: cannot find package "github.com/knative/build-pipeline/pkg/client/informers/externalversions/pipeline/v1alpha1" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/client/informers/externalversions/pipeline/v1alpha1 (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/client/informers/externalversions/pipeline/v1alpha1 (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/client/informers/externalversions/pipeline/v1alpha1 (from $GOPATH)
pkg/client/informers/externalversions/pipeline/v1alpha1/pipeline.go:24:2: cannot find package "github.com/knative/build-pipeline/pkg/client/listers/pipeline/v1alpha1" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/client/listers/pipeline/v1alpha1 (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/client/listers/pipeline/v1alpha1 (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/client/listers/pipeline/v1alpha1 (from $GOPATH)
pkg/reconciler/v1alpha1/pipeline/resources/dag.go:24:2: cannot find package "github.com/knative/build-pipeline/pkg/errors" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/errors (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/errors (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/errors (from $GOPATH)
pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go:28:2: cannot find package "github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/pipelinerun/resources" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/pipelinerun/resources (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/pipelinerun/resources (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/pipelinerun/resources (from $GOPATH)
pkg/reconciler/v1alpha1/taskrun/taskrun.go:44:2: cannot find package "github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/config" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/config (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/config (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/config (from $GOPATH)
pkg/reconciler/v1alpha1/taskrun/taskrun.go:45:2: cannot find package "github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/entrypoint" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/entrypoint (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/entrypoint (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/entrypoint (from $GOPATH)
pkg/reconciler/v1alpha1/taskrun/taskrun.go:46:2: cannot find package "github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/resources" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/resources (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/resources (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/resources (from $GOPATH)
test/controller.go:29:2: cannot find package "github.com/knative/build-pipeline/pkg/client/clientset/versioned/fake" in any of:
    /home/prow/go/src/github.com/knative/build/vendor/github.com/knative/build-pipeline/pkg/client/clientset/versioned/fake (vendor tree)
    /usr/local/go/src/github.com/knative/build-pipeline/pkg/client/clientset/versioned/fake (from $GOROOT)
    /home/prow/go/src/github.com/knative/build-pipeline/pkg/client/clientset/versioned/fake (from $GOPATH)
Exit program with code 0 regardless of the above error
```

Running `go test ./... -covermode=count -coverprofile /tmp/foo` from head works:

```
$ go test ./... -covermode=count -coverprofile /tmp/foo
?       github.com/knative/build-pipeline/cmd/controller        [no test files]
?       github.com/knative/build-pipeline/cmd/kubeconfigwriter  [no test files]
?       github.com/knative/build-pipeline/cmd/webhook   [no test files]
?       github.com/knative/build-pipeline/pkg   [no test files]
?       github.com/knative/build-pipeline/pkg/apis/pipeline     [no test files]
ok      github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1    0.341s  coverage: 19.5% of statements
?       github.com/knative/build-pipeline/pkg/client/clientset/versioned        [no test files]
?       github.com/knative/build-pipeline/pkg/client/clientset/versioned/fake   [no test files]
?       github.com/knative/build-pipeline/pkg/client/clientset/versioned/scheme [no test files]
?       github.com/knative/build-pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1        [no test files]
?       github.com/knative/build-pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1/fake   [no test files]
?       github.com/knative/build-pipeline/pkg/client/informers/externalversions [no test files]
?       github.com/knative/build-pipeline/pkg/client/informers/externalversions/internalinterfaces      [no test files]
?       github.com/knative/build-pipeline/pkg/client/informers/externalversions/pipeline        [no test files]
?       github.com/knative/build-pipeline/pkg/client/informers/externalversions/pipeline/v1alpha1       [no test files]
?       github.com/knative/build-pipeline/pkg/client/listers/pipeline/v1alpha1  [no test files]
ok      github.com/knative/build-pipeline/pkg/errors    0.496s  coverage: 100.0% of statements
?       github.com/knative/build-pipeline/pkg/logging   [no test files]
ok      github.com/knative/build-pipeline/pkg/reconciler        1.236s  coverage: 31.6% of statements
?       github.com/knative/build-pipeline/pkg/reconciler/testing        [no test files]
ok      github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/pipeline/resources    0.287s  coverage: 88.7% of statements
ok      github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/pipelinerun   0.302s  coverage: 89.0% of statements
ok      github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/pipelinerun/resources 0.405s  coverage: 89.7% of statements
ok      github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun       0.263s  coverage: 77.7% of statements
ok      github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/config        0.271s  coverage: 91.7% of statements
ok      github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/entrypoint    5.068s  coverage: 84.6% of statements
ok      github.com/knative/build-pipeline/pkg/reconciler/v1alpha1/taskrun/resources     0.475s  coverage: 86.5% of statements
?       github.com/knative/build-pipeline/pkg/system    [no test files]
ok      github.com/knative/build-pipeline/test  0.197s  coverage: 6.5% of statements
?       github.com/knative/build-pipeline/test/gohelloworld     [no test files]
```

/assign @steuhs

--- a/ci/prow/config.yaml
+++ b/ci/prow/config.yaml
@@ -1818,7 +1818,7 @@ periodics:
   decorate: true
   extra_refs:
   - org: knative
-    repo: build
+    repo: build-pipeline
     base_ref: master
     clone_uri: "https://github.com/knative/build-pipeline.git"
   spec:

